pyrules/rules.py

```diff
@@ -1,93 +1,161 @@
-        if translations:
-            translator = Translator(translations)
-            for rule in self.rules:
-                for key in rule.keys():
-                    rule[key] = [translator.replace(item) for item in rule[key]]
-    def should_trigger(self, context):
-        return True
+        if name:
+            self.name = name
+        self._current_ruleid = None
+
     def perform(self, context):
         count = 0
         for rule in self.rules:
-            if all([eval(condition, context.as_dict) for condition in rule['if']]):
+            evaluator = LogicEvaluator(
+                rule['if'].get('logic'), rule['if']['conditions'])
+            if evaluator.evaluate(context):
                 count = count + 1
-                self._current_ruleid = rule.get('id', count)
+                self._current_ruleid = rule.get('rule') or count
                 for action, target in zip(rule['then'], rule['target']):
-                    if context._translations:
-                        action = context._translations.replace(action)
-                        target = context._translations.replace(target)
-                    result = context[target.replace('context.', '').strip()] = eval(action, context.as_dict)
+                    #if context._translations:
+                    #    action = context._translations.replace(action)
+                    #    target = context._translations.replace(target)
+                    result = \
+                        context[target.replace('context.', '').strip()] = (
+                            eval(action, context.as_dict)
+                            if isinstance(action, basestring)
+                            else action)
                     self.record(context, result)
             else:
-                break
+                continue
         else:
             self._current_ruleid = None
             return True
         return False
+
     @property
     def ruleid(self):
         if self._current_ruleid:
-            return "%s.%s" % (super(TableRuleset, self).ruleid, self._current_ruleid)
-        return super(TableRuleset, self).ruleid
-    
+            return "%s.%s" % (super(TableRule, self).ruleid, self._current_ruleid)
+        return super(TableRule, self).ruleid
+
+    @classmethod
+    def from_yaml(cls, text):
+        return cls._from_data(yaml.load(text))
+        
+    @classmethod
+    def from_json(cls, text):
+        return cls._from_data(json.loads(text))
+
+    @classmethod
+    def _from_data(cls, data):
+        # We have to convert non-string data in clauses back to strings,
+        # because they will be eval-ed
+        rules = []
+        for rule in data['rules']:
+            obj = {
+                'rule': rule.get('rule'),
+                'then': rule['then'],
+                'target': rule['target']}
+            if_clause = {}
+            # Convert conditions to dictionaries, i.e. "foo" becomes {"foo": True}
```

Comment:
we should be able to apply translations here (#20) and everything else should work just fine from here onwards I think